hashes/src/util.rs

```diff
@@ -3,15 +3,23 @@
 #[macro_export]
 /// Adds hexadecimal formatting implementation of a trait `$imp` to a given type `$ty`.
 macro_rules! hex_fmt_impl(
-    ($reverse:expr, $ty:ident) => (
-        $crate::hex_fmt_impl!($reverse, $ty, );
+    ($reverse:expr, $len:expr, $ty:ident) => (
+        $crate::hex_fmt_impl!($reverse, $len, $ty, );
     );
-    ($reverse:expr, $ty:ident, $($gen:ident: $gent:ident),*) => (
+    ($reverse:expr, $len:expr, $ty:ident, $($gen:ident: $gent:ident),*) => (
         impl<$($gen: $gent),*> $crate::_export::_core::fmt::LowerHex for $ty<$($gen),*> {
             #[inline]
             fn fmt(&self, f: &mut $crate::_export::_core::fmt::Formatter) -> $crate::_export::_core::fmt::Result {
                 if $reverse {
-                    $crate::_export::_core::fmt::LowerHex::fmt(&self.0.backward_hex(), f)
+                    use $crate::Hash as _;
+
+                    let mut backwards = [0_u8; $len];
+                    for (i, b) in self.as_byte_array().iter().rev().enumerate() {
+                        backwards[i] = *b;
```

Comment:
There's a `reverse` method which is also more efficient but I still don't get why you can't just call `fmt_hex_exact!(f, $len, self.as_byte_array().iter().rev(), Case::Lower)` which is even more efficient.

Also FYI, the original design was specifically so that the consumers don't have to depend on an unstable hex stuff (but there was an oversight with errors).

Reply:
Used `fmt_hex_exact!`.